File: platform-core-service/platform-core-service.Common/Models/DTOs/EntityDTO/CommunityMember/SelectCommunityMembershipRequestDTO.cs
using platform_core_service.Common.Entities.BaseEntity;
using platform_core_service.Common.Models.DTOs.EntityDTO.Profile;

namespace platform_core_service.Common.Models.DTOs.EntityDTO.CommunityMember
{
    public class SelectCommunityMembershipRequestDTO : IBaseKey<string>
    {
        public string Id { get; set; } = null!;
        public string CommunityId { get; set; } = null!;
        public string RequesterId { get; set; } = null!;
        public SelectProfileDTO? Requester { get; set; }
        public DateTimeOffset? DateCreated { get; set; }
        public bool HasBlockedRelation { get; set; }
        public bool IsProfileRestricted { get; set; }
        public string? RestrictedMessage { get; set; }
        public bool CanApprove { get; set; } = true;
        public bool CanReject { get; set; } = true;
    }
}

File: frontend/src/types/community-requests/select-community-membership-request-dto.ts
import { SelectProfileDTO } from "../profile/select-profile-dto";

export interface SelectCommunityMembershipRequestDTO {
    id: string;
    communityId: string;
    requesterId: string;
    requester?: SelectProfileDTO;
    dateCreated?: string;
    hasBlockedRelation?: boolean;
    isProfileRestricted?: boolean;
    restrictedMessage?: string | null;
    canApprove?: boolean;
    canReject?: boolean;
}
